src/tries/mod.rs

```diff
@@ -75,6 +71,38 @@ impl Filter for Node {
         curr.append_inner(&mut nums);
         nums
     }
+
+    fn find_prefix_case_insensitive(&self, s: &str) -> HashSet<u32> {
```

Comment:
The recursion is very expensive. We are doing that in line 75. It will be a good move to remove that and stay inside the loop, instead of the recursion.